backend/src/main/java/com/google/sps/servlets/SearchMatchServlet.java

```diff
@@ -116,20 +127,25 @@ private boolean isExpired(Participant participant) {
   }
 
   /** Send JSON response for expired participant that has been removed from datastore */
-  private void sendExpiredResponse(HttpServletResponse response) throws IOException {
+  private void sendExpiredResponse(HttpServletResponse response, Participant participant)
+      throws IOException {
     JSONObject expired = new JSONObject();
     expired.put(JSON_MATCH_STATUS, "expired");
-
+    try {
+      emailNotifier.sendExpiredEmail(participant.getUsername());
+    } catch (MessagingException e) {
+      e.printStackTrace();
+    }
     // Send the JSON back as the response
     response.setContentType("application/json");
     response.getWriter().println(expired.toString());
   }
 
   /** Send JSON response for no match yet */
-  private void sendNoMatchResponse(HttpServletResponse response) throws IOException {
+  private void sendNoMatchResponse(HttpServletResponse response, Participant participant)
```

Comment:
Do we need a sendNoMatchEmail() function for this one?

Reply:
No we do not !

Reply:
If we don't need to send email, why do we need to pass Participant? i.e we don't need this change.

Reply:
Fixed